ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp

```diff
@@ -637,8 +589,47 @@ using TBase = TComputationValue<TSpillingSupportState<Sort, HasCount>>;
         return true;
     }
 
+    NUdf::TUnboxedValue* ExtractInMemory() {
+        if (Full.empty())
+            return nullptr;
+
+        const auto ptr = Full.back();
+        Full.pop_back();
+        return static_cast<NUdf::TUnboxedValue*>(ptr);
+    }
+
+    void SealInMemory() {
+        if constexpr (!HasCount) {
+            static_assert (Sort);
+            // Remove placeholder for new data
+            Storage.resize(Storage.size() - Indexes.size());
+
+            Full.reserve(Storage.size() / Indexes.size());
+            for (auto it = Storage.begin(); it != Storage.end(); it += Indexes.size()) {
+                Full.emplace_back(&*it);
+            }
+
+            std::sort(Full.rbegin(), Full.rend(), LessFunc);
+            return;
+        }
+
+        Free.clear();
+        Free.shrink_to_fit();
```

Comment:
Cppref says that `shrink_to_fit` is not obliged to actually shrink:
https://en.cppreference.com/w/cpp/container/vector/shrink_to_fit

As far as I know the only way to completely clear container is to swap it with empty container

Reply:
Hm, I kept the previous implementation.
I think we should be fine for now: https://stackoverflow.com/questions/23502291/is-shrink-to-fit-the-proper-way-of-reducing-the-capacity-a-stdvector-to-its/23503995#23503995